File: 0136-single-number/0136-single-number.c
int singleNumber(int* nums, int numsSize) {
    int answer = 0;

    for (int i = 0; i < numsSize; i++) {
        // 같은 숫자는 XOR하면 0이 되고,
        // 0과 XOR한 값은 자기 자신이 된다.
        answer ^= nums[i];
    }

    return answer;
}

File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        answer = 0

        # 모든 숫자를 XOR하면 두 번 등장한 숫자는 사라지고
        # 한 번만 등장한 숫자만 남는다.
        for num in nums:
            answer ^= num

        return answer